manifests/params.pp

```diff
@@ -253,6 +253,8 @@
 
   # Application logging
   $logging_level = 'info'
+  $logging_type = 'file'
```

Comment:
This is indeed the most sane default - everything stays the same for production instance.

But logging to files is considered bad practice, except some corner cases (e.g. apache httpd access logs) using standard logging mechanisms should be preferred. I was hoping to default to journald (if present), syslog (otherwise) and just dropping `/etc/rsyslog.d/foreman.conf` configuration file and alerting daemon to reload (similarly for syslog-ng). If that's too complex in Puppet, we can just leave file logging for now.

Reply:
> This is indeed the most sane default - everything stays the same for production instance.

That was indeed exactly what I thought, also because I can't quite oversee what else is affected. Start with configuration options and safe defaults.

If journald is just drop in, we could use facts to determine if systemd is used. I'm hesitant to start managing syslog since that can be fairly complex and more up to the admin to set up. That's why I'd fall back to `file` if systemd is not used. My plan was to revisit the defaults later.

Reply:
The question is how or when do we find out it's now safe to change the default, too bad didn't have this in the beginning of 1.18 dev window. Would it make sense to change the default just after we branch out?

I agree that we shouldn't manage syslog but dropping a oneliner to `/etc/rsyslog.d/foreman.conf` would be acceptable I think?

Reply:
> The question is how or when do we find out it's now safe to change the default, too bad didn't have this in the beginning of 1.18 dev window. Would it make sense to change the default just after we branch out?

For me it depends on whether journald is a drop in replacement:
* Does journald also log to a file?
* Can foreman-debug extract logs from journald?

> I agree that we shouldn't manage syslog but dropping a oneliner to /etc/rsyslog.d/foreman.conf would be acceptable I think?

The difficulty is that we can't rely on any syslog implementation. I can imagine users are using `syslog-ng` for example. Then we also need to support that. Suddenly we have a few extra parameters for things that IMHO belong in a profile module. Will we manage the syslog daemon as well for example. We've worked hard to keep that out of puppet-foreman. It just doesn't fit the design. I really don't want to touch syslog at all.

Reply:
Unfortunately, journald main idea is all around binary journal that can be sealed - there is no file logging in journald. It can however forward to syslog and that's what it does by default on most systems these days.

I will create patch for foreman-debug to collect the logs from there for sure.

If we don't want to touch syslog, then another option is to ditch `/var/log/foreman/production.log` completely. An installer could archive/rename all files in pattern `production*.log*` and drop a dummy file with short explanation that this is not used. I don't think this is a bad experience for users after upgrade at all. Things change.

Structured logging patch is not only about structured logging, but also about collecting things from other parties like backend systems or smart proxy. With files, we can't achieve that, but having things in journald allows us to start working towards this goal.

Reply:
To me it sounds like we should talk to our users about their expectations because nowadays I'm somewhat removed from actual deployments. Should we open a topic on community.theforeman.org?

Reply:
Yep, started it: https://community.theforeman.org/t/default-logging-in-foreman-file-syslog-or-journald/8675